[N Vijay Narayanan] iP

myc37 opened the pull request:
# Innkeeper 
:wave: **_Greetings Travellers!_** :wave:

Isn't it such a hassle to keep track of all the tasks in your busy schedule?
Fret not! The innkeeper is here to be your friend and lend a helping hand :handshake:

> A friend in need is a friend indeed [(Source)](www.theidioms.com)

The innkeeper is a personal assistant chatbot that helps you keep track of all your tasks.
It is,
- Text-based :abcd:
- Easy to learn and use :baby:
- Free! :money_mouth_face:

## How to get started 
1. Download the executable .jar file [here](https://github.com/myc37/ip/releases/tag/A-Jar).
2. Double click to open it.
3. Type guide to familiarise yourself with all valid commands.
4. Interact with the Innkeeper!

## Features
- [x] Add tasks
- [x] Mark tasks as complete or unmark them
- [x] Remove tasks
- [x] View all tasks
- [x] View all tasks on a specific date
- [x] Find tasks via matching keyword
- [x] Save tasklist in harddisk
- [ ] Reminders for upcoming tasks

## Code Preview
Here's a quick peek at the main method! :smile:
```java
public static void main(String[] args) {
        Ui innkeeper = new Ui();
        TaskList taskList = new TaskList();
        Storage storage = new Storage(SAVE_FILE_DIRECTORY, SAVE_FILE_NAME);

        try {
            storage.loadData(taskList);
        } catch (ChatBotException e) {
            innkeeper.error(e.getMessage());
        }

        Parser parser = new Parser(innkeeper, storage, taskList);
        Scanner sc = new Scanner(System.in);
        Boolean loop = true;

        innkeeper.greet(taskList.isEmpty());
        while (loop.equals(true)) {
            innkeeper.prompt();
            String rawInput = sc.nextLine();
            loop = parser.parse(rawInput);
        }
        sc.close();
    }
```

Commit:
Add Gradle support

myc37 committed:
Level 1

myc37 committed:
Level 2

myc37 committed:
Level-3 completed, added mark/unmark feature with some basic error handling, created new Task class and updated TaskList class to use an array of Tasks

myc37 committed:
Level-4 completed, child classes for Task implemented and some placeholders put in for error handling to improved in Level-5

myc37 committed:
A-TextUiTesting initialized, currently have a set of tests for all valid commands, will include error handling tests after Level-5

myc37 committed:
Level 5 completed, created ChatBotException class and handled most (if not all) possible errors, also created new testcases for error handling

myc37 committed:
Level-6 complete, changed task list data structure from array to arraylist, implemented delete function and updated test cases to include delete

myc37 committed:
Level-6 complete, changed task list data structure from array to arraylist, implemented delete function and updated test cases to include delete

myc37 committed:
Level-7 completed, tasks will be saved in save file(src/data/data.txt) upon any change to tasklist and will be loaded from save file when ChatBot is started up

myc37 committed:
Level-8 completed, created a new ChatBotDateTime class to handle dates and timestamps for deadlines and events, updated guide accordinly

myc37 committed:
Merge branch 'branch-Level-8'

myc37 committed:
A-MoreOOp completed, no additional code but refactored code by adding Ui, Storage and Parser classes

myc37 committed:
A-Packages completed, seperated .java files into appropriate packages

myc37 committed:
A-JUnit completed, created test classes for TaskList as well as Timestamp for unit testing

myc37 committed:
A-Jar completed, packaged app as a JAR file

myc37 committed:
A-JavaDoc completed, added JavaDocs for all classes

myc37 committed:
Complete A-CodingStandard

myc37 committed:
Complete Level-9

myc37 committed:
Merge branch 'branch-A-JavaDoc'

myc37 committed:
Merge branch 'branch-A-CodingStandard'

myc37 committed:
Merge branch 'branch-Level-9'

myc37 committed:
Fix compiler path in runtest.bat

myc37 committed:
Update JUnit test cases and bugfix for delete function

myc37 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

myc37 committed:
Complete A-Gradle

myc37 committed:
Merge branch 'branch-A-Gradle'

myc37 committed:
Complete A-Checkstyle

myc37 committed:
Merge branch 'branch-A-Checkstyle'

myc37 committed:
Add basic GUI

myc37 committed:
Complete Level-10

myc37 committed:
Merge branch 'branch-Level-10'

myc37 committed:
Update GUI

aidanaden commented on src/main/java/chatbot/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+     * Instantiates a new Parser.
+     *
+     * @param ui The UI.
+     * @param storage   The storage.
```
Perhaps comments can be more descriptive of what UI/storage/tasklist is or contains? :)

aidanaden commented on src/main/java/chatbot/util/Parser.java:
```diff
@@ -0,0 +1,201 @@
+        String[] input = rawInput.split(" ");
+        String response;
+        switch (input[0]) {
+        case "bye":
```
switch-case follows style guide 👍

aidanaden commented on src/main/java/chatbot/util/Storage.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    /**
+     * Load data.
```
maybe comments can be more specific? what does 'data' mean here? 🤔

aidanaden commented on src/main/java/chatbot/util/Ui.java:
```diff
@@ -0,0 +1,110 @@
+ */
+public class Ui {
+
+    private static final String[] GREETING_QUOTES = {
```
Could this be stored as an enum perhaps?

aidanaden commented on src/main/java/chatbot/task/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+     * Instantiates a new Deadline.
+     *
+     * @param title The title of the deadline.
+     * @param by    The timestamp of the deadline.
```
The spaces between the parameter description and the parameter name seems to be inconsistent, based on style guide I think description should be a single space away from the parameter name - https://se-education.org/guides/conventions/java/intermediate.html#comments

aidanaden reviewed (commented):
LGTM, just few style guide nits to fix 👍

myc37 committed:
Refactor Main.java

There are three images used for the GUI. The user and innkeeper icons are set in MainWindow.java controller but background image is set in MainWindow.fxml by passing its url to the style prop of dialogContainer.

When the application is packaged into a .jar file and executed, the two icons appear fine but the background image does not load.

Let's,
* use getResourceAsStream() to retrieve background image in MainWindow.java controller
* set the background of dialogContainer in the initialize() method
* remove styles related to setting background of dialogBox in MainWindow.fxml

The relative URL to the background image becomes incorrect when the application is packaged into a .jar file. Moving the code for setting the background image from MainWindow.fxml to MainWindow.java allows for the reference to the background image to persist in the .jar file.

myc37 committed:
Complete A-CodeQuality

myc37 committed:
Merge pull request #1 from myc37/branch-A-CodeQuality

Complete A-CodeQuality

myc37 committed:
Add Contact.java

myc37 committed:
Add ChatBotList.java, refactor Tasklist.java

myc37 committed:
Add ContactList.java

myc37 committed:
Complete D-Contacts

myc37 committed:
Merge pull request #2 from myc37/branch-D-Contacts

Branch d contacts

myc37 committed:
Complete A-CI

myc37 committed:
Remove data directories from remote repo

myc37 committed:
Remove unused files from remote repo

myc37 committed:
Complete A-Assert

myc37 committed:
Merge pull request #3 from myc37/branch-A-Assert

Complete A-Assert

myc37 committed:
Fix GUI scrolling issue

myc37 committed:
Add UI.png

myc37 committed:
Add User Guide

myc37 committed:
Set theme jekyll-theme-minimal

myc37 committed:
Complete A-Release

myc37 committed:
Merge branch 'master' of https://github.com/myc37/ip

myc37 committed:
Fix README issues

myc37 committed:
Fix abstraction issues

myc37 committed:
Fix header comments
